app/src/main/java/com/googleresearch/capturesync/SoftwareSyncController.java

```diff
@@ -114,9 +114,9 @@
     public SoftwareSyncController(
             MainActivity mainActivity, PhaseAlignController phaseAlignController, PeriodCalculator periodCalculator) {
         mMainActivity = mainActivity;
-        mSoftwareSyncUtils = mainActivity.getApplicationInterface().getSoftwareSyncUtils();
         mPhaseAlignController = phaseAlignController;
         mPeriodCalculator = periodCalculator;
+        mSoftwareSyncUtils = new SoftwareSyncUtils(mainActivity, this);
```

Comment:
Why your utils class isn't static?

Reply:
Fistly, it is needed only when RecSync is running, and secondly, if it would be static, we'll have to pass `MainActivity` to virtually all of its methods.

Reply:
Then it is not utils. I think)

Reply:
Activity is a class that can be changed as a part of not our code (OS and so on), so providing links to it anywhere isn't a great idea. It means that you can use some other names --- SoftwareSyncHelper or smth else if you have to store activity and use this class only during controller lifecycle